doc-tool/test/JavaConverterTest.scala

```diff
@@ -3,109 +3,317 @@ package dottydoc
+      def typeParams = "String" :: "String" :: Nil
+      def constructors = List(List(paramList))
+      def superTypes = new NoLink("title", "query") :: Nil
+      def members = Nil
+      def companionPath = "path" :: "to" :: "companion" :: Nil
+      def companionPath_=(xs: List[String]) = Unit
     }
-    trait TestCompanion extends TestEntity with Companion {
+    assertSerializedCorrectly(caseClass, caseClass.asJava())
+    val obj = new EObject {
+      def symbol = NoSymbol
+      def name = "someObject"
+      def path = "path" :: "to" :: "object" :: Nil
+      def comment = None
+      def annotations = List("test")
+      def parent = NonEntity
+      def modifiers = "protected" :: Nil
+      def typeParams = "String" :: "String" :: Nil
+      def superTypes = new NoLink("title", "query") :: Nil
+      def members = df :: Nil
       def companionPath = "path" :: "to" :: "companion" :: Nil
       def companionPath_=(xs: List[String]) = Unit
     }
-    trait TestReturnValue extends TestEntity with ReturnValue {
-      def returnValue =
-        new TypeReference("String", new NoLink("title", "target"), List())
+    assertSerializedCorrectly(obj, obj.asJava())
+    val typeAlias = new TypeAlias {
+      def symbol = NoSymbol
+      def name = "typeAlias"
+      def path = "path" :: "to" :: "typeAlias" :: Nil
+      def comment = None
+      def annotations = List("test")
+      def parent = NonEntity
+      def modifiers = "private" :: Nil
+      def typeParams = "String" :: "String" :: Nil
+      def alias = Some(new TypeReference("String", new NoLink("title", "target"), List()))
     }
-    trait TestImplicitlyAddedEntity
-        extends TestEntity
-        with ImplicitlyAddedEntity {
-      def implicitlyAddedFrom =
-        Some(
-          new TypeReference("String", new NoLink("title", "target"), List()))
+    assertSerializedCorrectly(typeAlias, typeAlias.asJava())
+    val vl = new Val {
+      val kind = "val"
+      def symbol = NoSymbol
+      def name = "val"
+      def path = "path" :: "to" :: "val" :: Nil
+      def comment = None
+      def annotations = List("test")
+      def parent = NonEntity
+      def modifiers = "private" :: Nil
+      def returnValue = new TypeReference("String", new NoLink("title", "target"), List())
+      def implicitlyAddedFrom = Some(
+        new TypeReference("String", new NoLink("title", "target"), List()))
+    }
+    assertSerializedCorrectly(vl, vl.asJava())
+    val pkg = new Package {
+      def symbol = NoSymbol
+      def name = "test"
+      def path = "path" :: "to" :: "test" :: Nil
+      def comment = None
+      def annotations = List("test")
+      def parent = NonEntity
+      def members = trt :: typeAlias :: Nil
+      def superTypes = new NoLink("title", "query") :: Nil
+    }
+    assertSerializedCorrectly(pkg, pkg.asJava())
+  }
+
+  def assertEach[E, A, C[E] <: Seq[E]](expected: C[E], serialized: Any, pairwiseAssertion: (E, A) => Unit) {
```

Comment:
Nope, looks like we don't! Removing the type annotations significantly improves the readability of the calls.

Can I amend the commit, or would that confuse the system?